frontend/packages/ceph-storage-plugin/src/components/ocs-install/existing-cluster-modal.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { Trans, useTranslation } from 'react-i18next';
+import { match as RouteMatch } from 'react-router';
+import { Button } from '@patternfly/react-core';
+import { Modal } from '@console/shared/src/components/modal';
+import { ClusterServiceVersionModel } from '@console/operator-lifecycle-manager/src';
+import { history, resourcePathFromModel } from '@console/internal/components/utils';
+import { ListKind, referenceForModel } from '@console/internal/module/k8s';
+import { StorageClusterKind } from '../../types';
+import { OCSServiceModel } from '../../models';
+
+export const ExistingClusterModal: React.FC<ExistingClusterModalProps> = ({
+  match,
+  storageCluster,
+}) => {
+  const {
+    params: { ns, appName },
+  } = match;
+  const { t } = useTranslation();
+  const [isOpen, setIsOpen] = React.useState(true);
+
+  const clusterName = storageCluster.items.find((sc) => sc.status.phase !== 'Ignored').metadata
+    .name;
```

Comment:
```suggestion
  const clusterName = storageCluster.items.find((sc) => sc.status.phase !== 'Ignored')?.metadata
   ?.name;
```

Reply:
This component will be called only when a cluster is present